### ISSUE TITLE
Add Calispel?

### ISSUE THREAD

**Issue by Ambrevar** [CodyReichert/awesome-cl#290]
GitHub: https://github.com/hawkir/calispel

I believe it provides a couple of procedures that lparallel misses, e.g. the "select" procedure to choose the next available channel among a set of channels.

**Comment by vindarel** [comment]
Maybe… at the moment I see 3 commits and zero stars… you want to post it to reddit to see if there is feedback, and giving yours?

**Comment by Ambrevar** [comment]
I believe this it's one of those older libraries that was re-hosted on GitHub.
Hence the small number of commits.

**Comment by Ambrevar** [comment]
I've finally found the time to test this.
I've used lparallel for some time as well as ChanL.

- Unlike lparallel, it does not require this initialization of a
  global kernel. so it works out of the box.

- Unlike lparallel, sending to a channel is a blocking operation (this
  makes lparallel unusable for proper CSP if I understand correctly).

- It has a blocking SELECT statement (called `fair-alt`), unlike
lparallel which has no such statement, and unlike ChanL which has only a
a non-blocking SELECT statement (which makes it almost useless).

- Lparallel has a weird syntax for sending to a channel (`submit-task` takes a function which is executed concurrently), which departs from usual CSP style.

In short, Calispel seems to just work for everything related to CSP.

If this verifies, then I would recommend Calispel over Lparallel and ChanL which have proven a bit painful for me because of the above points.

**Comment by Ambrevar** [comment]
Reddit discussion: https://old.reddit.com/r/Common_Lisp/comments/jste48/lparallel_vs_chanl_vs_calispel/?

**Comment by Ambrevar** [comment]
Another benefit of Calispel: FAIR-ALT (the 'select' statement in CSP) is an expression and returns a value, unlike ChanL.

**Comment by Ambrevar** [comment]
Looks like no one really had an opinion about CSP on Reddit.

To test Calispel more in depth, I've replaced ChanL with it in Nyxt: https://github.com/atlas-engineer/nyxt/pull/1047

Works like a charm, and I'm already benefiting from `fair-alt`!

To conclude, I suggest that we put this library at the top of the recommendations on concurrency.  Maybe hand in hand with lparallel, since the latter has convenience functions like `pmap`.

**Comment by vindarel** [comment]
OK. We can put it above ChanL, if it has more features. I think a link to the reddit discussion, or a sentence with your opinion would be beneficial. Not only the reference, but something more opinionated, that gives more pointers.

(lparallel is far more used so it should still stand out, IMO)

**Comment by Ambrevar** [comment]
What about this:

"Calispel is a complete is a complete implementation of
[CSP](https://en.wikipedia.org/wiki/Communicating_sequential_processes)
with blocking, optionally buffered channels and a "CSP select"
statement.

It's complete, flexible and easy to use."

**Comment by Ambrevar** [comment]
Thanks again! :)
